api/src/routes/tests/http-test.js

```diff
@@ -47,6 +50,246 @@ describe('API HTTP Route tests', function describeTest() {
+            const [k, v] = kv.split('=');
+            return { ...acc, [k]: v };
+          }, {});
+
+          // expect(refreshTokenCookie.Domain).to.equal('localhost');
+          // expect(refreshTokenCookie.Path).to.equal('/');
+          expect(Object.keys(refreshTokenCookie).includes('HttpOnly')).to.equal(true);
+
+          expect(moment(refreshTokenCookie.Expires).isBefore(moment().add(7, 'days'))).to.equal(true);
+          expect(moment(refreshTokenCookie.Expires).isAfter(moment().add(7, 'days').add(-3, 's'))).to.equal(true);
+
+          const refreshToken = refreshTokenCookie.refresh_token_organisation_561a679c0c5d017e4004714f;
+          const decodedRefreshToken = jsonwebtoken.verify(refreshToken, process.env.APP_SECRET);
+
+          expect(decodedRefreshToken.userId).to.equal('561a679c0c5d017e4004714f');
+          expect(decodedRefreshToken.tokenId).to.equal('561a679c0c5d017e4004714f');
+          expect(decodedRefreshToken.tokenType).to.equal('organisation_refresh');
+          expect(moment.unix(decodedRefreshToken.exp).isBefore(moment().add(7, 'days'))).to.equal(true);
+          expect(moment.unix(decodedRefreshToken.exp).isAfter(moment().add(7, 'days').add(-3, 's'))).to.equal(true);
+
+          // Test body (access token)
+          const decodedAccessToken = jsonwebtoken.verify(res.text, process.env.APP_SECRET);
+          expect(decodedAccessToken.userId).to.equal('561a679c0c5d017e4004714f');
+          expect(decodedAccessToken.tokenType).to.equal('organisation');
+          expect(decodedAccessToken.tokenId).to.equal('561a679c0c5d017e4004714f');
+
+          expect(moment.unix(decodedAccessToken.exp).isBefore(moment().add(1, 'h'))).to.equal(true);
+          expect(moment.unix(decodedAccessToken.exp).isAfter(moment().add(1, 'h').add(-3, 's'))).to.equal(true);
+        });
+    });
+
+    it('should return 401 when user access token is invalid', async () => {
+      await apiApp
+        .post(routes.AUTH_JWT_ORGANISATION)
+        .expect(401)
+        .expect((res) => {
+          expect(res.headers['set-cookie']).to.equal(undefined);
+        });
+    });
+  });
+
+  describe('/api/auth/jwt/refresh', () => {
+    it('should return 200 and user access token when user refresh token cookie is valid', async () => {
+      let refreshTokenCookie = '';
+      let accessToken = '';
+
+      await apiApp
+        .post(routes.AUTH_JWT_PASSWORD)
+        .auth('testy@mctestface.com', 'password1')
+        .expect((res) => {
+          refreshTokenCookie = res.headers['set-cookie'][0];
+          accessToken = res.text;
+        });
+
+      // To confirm `iat` and `exp` are changed
+      await delay(1000);
+      refreshTokenCookie = refreshTokenCookie.split('; ')[0];
+
+      await apiApp
+        .post(routes.AUTH_JWT_REFRESH)
+        .set('Cookie', refreshTokenCookie)
+        .send({ id: '561a679c0c5d017e4004714f', tokenType: 'user' })
+        .expect(200)
+        .expect('Content-Type', /text\/html/)
```

Comment:
I think it might be odd to return `text/html` as the content type because the response is actually just the token as text rather than html. What do you think @h-kanazawa?

Reply:
Perhaps `text/plain`?

Reply:
Wait, isn't it `application/json`? I'm guessing that Content-Type is there for a reason though....

Reply:
Either `application/json` or `text/plain` would be fine I think. Yeah I'm unsure why it's `text/html` because I think it's just text.

Reply:
I add 3 env variables

- `REFRESH_TOKEN_EXPIRATION_MSEC` (for cookie's expires and maxAge) e.g. "3600000"
- `REFRESH_TOKEN_EXPIRATION`(for jwt exp) e.g. "7d" This value is expected to be same length with `REFRESH_TOKEN_EXPIRATION_MSEC`
- `ACCESS_TOKEN_EXPIRATION` (for jwt exp) e.g. "1h"

Reply:
I changed it to text/plain.